### PR TITLE
Docs: Update C# SDK docs page

### LINKED ISSUE #1769: Updated C# documentation to use NuGet package
**Is your feature request related to a problem? Please describe.**

Now that #1753 has been finished and we have a [1.8.0](https://www.nuget.org/packages/AgonesSDK/1.8.0) Nuget package, we should update the [C# SDK Documentation](https://agones.dev/site/docs/guides/client-sdks/csharp/) to use this package.

**Describe the solution you'd like**

Updated documentation, that is appropriate for C# (I've no idea 😃)

**Describe alternatives you've considered**

Leave it alone, since it works, but that's not ideal for C# users.

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by Reousa**
/kind documentation

@markmandel I'm unsure where to place the new Alpha docs, on the same page or a different one?

Resolves #1769

### CODE REVIEW

**Comment by aLekSer** [comment]
A new E2E test fail in less than one second.
```
go test -mod=vendor -race -parallel=64 agones.dev/agones/test/e2e --kubeconfig /root/.kube/config \
	--gameserver-image=gcr.io/agones-images/simple-game-server:0.1 \
	--pullsecret= \
	--feature-gates="" \
	--namespace=""
time="2020-09-13 12:38:35.991" level=info msg="Starting e2e test(s)" featureGates="ContainerPortAllocation=true&Example=true&PlayerTracking=false&SDKWatchSendOnExecute=false" gameServerImage="gcr.io/agones-images/simple-game-server:0.1" namespace= perfOutputDir= pullSecret= stressTestLevel=0 version=
time="2020-09-13 12:38:36.447" level=info msg="cause for namespace deletion error" reason=NotFound
time="2020-09-13 12:38:36.447" level=error msg="failed to cleanup e2e namespaces" error="namespaces \"1600000025\" not found"
FAIL	agones.dev/agones/test/e2e	0.530s
FAIL
```

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 1]
```diff
@@ -25,10 +33,14 @@ Download the source {{< ghlink href="sdks/csharp" >}}directly from GitHub{{< /gh
 
 Reference the SDK in your project & create a new instance of the SDK wrapper:
 
+### Initialization
```
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:36]: Might be a good time to ask - will there be any kind of `import` statement that needs to happen at the top of the file -- might be good to include that here?

Comment (Reousa) [reply site/content/en/docs/Guides/Client SDKs/csharp.md:36]: Yeah, will get it updated 👌

**Comment by markmandel** [comment]
Updating with Master - there's a clog in CI that's been fixed.

**Review by markmandel — APPROVED** [review]
:+1:

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 2]
```diff
@@ -85,11 +112,62 @@ This process is non-blocking internally.
 agonesSDK.WatchGameServer((gameServer) => { Console.WriteLine($"Server - Watch {gameServer}");});
 ```
 
+### Alpha: PlayerConnect
```
Comment (markmandel) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:115]: Just realised one thing - let's make this match the REST API

Let's put a "Player Tracking" header of the top, and include the `{{< alpha title="Player Tracking" gate="PlayerTracking" >}}` standard shortcode, so we include the standard warning

**Review by markmandel — CHANGES_REQUESTED** [review]

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 3]
```diff
@@ -85,11 +112,62 @@ This process is non-blocking internally.
 agonesSDK.WatchGameServer((gameServer) => { Console.WriteLine($"Server - Watch {gameServer}");});
 ```
 
+### Alpha: PlayerConnect
+
+This method increases the SDK’s stored player count by one, and appends this playerID to GameServer.Status.Players.IDs.
+Returns true and adds the playerID to the list of playerIDs if the playerIDs was not already in the list of connected playerIDs.
+
+```csharp
+bool ok = PlayerConnectAsync(playerId);
```
Comment (yoshd) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:121]: Thank you for writing the documentation!
I suggest the same as calling the actual method.
```suggestion
bool ok = await agones.Alpha().PlayerConnectAsync(playerId);
```

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 4]
```diff
@@ -85,11 +112,62 @@ This process is non-blocking internally.
...
+```
+
+### Alpha: PlayerDisconnect
+
+This function decreases the SDK’s stored player count by one, and removes the playerID from GameServer.Status.Players.IDs.
+Will return true and remove the supplied playerID from the list of connected playerIDs if the playerID value exists within the list.
+
+```csharp
+bool ok = PlayerDisconnectAsync(playerId);
```
Comment (yoshd) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:130]: This is also the same

```suggestion
bool ok = await agones.Alpha().PlayerDisconnectAsync(playerId);
```

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 5]
```diff
@@ -85,11 +112,62 @@ This process is non-blocking internally.
...
+```
+
+### Alpha: SetPlayerCapacity
+
+Update the `GameServer.Status.Players.Capacity` value with a new capacity.
+
+```csharp
+var status = await SetPlayerCapactyAsync();
```
Comment (yoshd) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:138]: This is also the same

```suggestion
var capacity = 100;
var status = await agones.Alpha().SetPlayerCapacityAsync(capacity);
```

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 6]
```diff
@@ -85,11 +112,62 @@ This process is non-blocking internally.
...
+```
+
+### Alpha: GetPlayerCapacity
+
+This function retrieves the current player capacity `GameServer.Status.Players.Capacity`. 
+This is always accurate from what has been set through this SDK, even if the value has yet to be updated on the GameServer status resource.
+
+```csharp
+long cap = await GetPlayerCapacityAsync();
```
Comment (yoshd) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:147]: This is also the same

```suggestion
long cap = await agones.Alpha().GetPlayerCapacityAsync();
```

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 7]
```diff
@@ -85,11 +112,62 @@ This process is non-blocking internally.
...
+```
+
+### Alpha: GetPlayerCount
+
+Returns the current player count
+
+```csharp
+long count = await GetPlayerCountAsync();
```
Comment (yoshd) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:155]: This is also the same

```suggestion
long count = await agones.Alpha().GetPlayerCountAsync();
```

File: site/content/en/docs/Guides/Client SDKs/csharp.md [hunk 8]
```diff
@@ -85,11 +112,62 @@ This process is non-blocking internally.
...
+```
+
+### Alpha: IsPlayerConnected
+
+This returns if the playerID is currently connected to the GameServer.
+This is always accurate, even if the value hasn’t been updated to the GameServer status yet.
+
+```csharp
+bool isConnected = IsPlayerConnectedAsync();
```
Comment (yoshd) [inline site/content/en/docs/Guides/Client SDKs/csharp.md:164]: This is also the same

```suggestion
var playerId = "player1";
bool isConnected = await agones.Alpha().IsPlayerConnectedAsync(playerId);
```

**Review by markmandel — APPROVED** [review]
Nice!

**Pull request merged**